backend/zoho/zoho.go

```diff
@@ -0,0 +1,1240 @@
+		ExtraHeaders: map[string]string{"Accept": "application/vnd.api+json"},
+		Parameters:   url.Values{},
+	}
+	opts.Parameters.Set("page[limit]", strconv.Itoa(10))
+	offset := 0
+OUTER:
+	for {
+		opts.Parameters.Set("page[offset]", strconv.Itoa(offset))
+
+		var result api.ItemList
+		var resp *http.Response
+		err = f.pacer.Call(func() (bool, error) {
+			resp, err = f.srv.CallJSON(ctx, &opts, nil, &result)
+			return shouldRetry(resp, err)
+		})
+		if err != nil {
+			return found, errors.Wrap(err, "couldn't list files")
+		}
+		if len(result.Items) == 0 {
+			break
+		}
+		for i := range result.Items {
+			item := &result.Items[i]
+			if item.Attributes.IsFolder {
+				if filesOnly {
+					continue
+				}
+			} else {
+				if directoriesOnly {
+					continue
+				}
+			}
+			item.Attributes.Name = f.opt.Enc.ToStandardName(item.Attributes.Name)
+			if fn(item) {
+				found = true
+				break OUTER
+			}
+		}
+		offset += 10
+	}
+	return
+}
+
+// List the objects and directories in dir into entries.  The
+// entries can be returned in any order but should be for a
+// complete directory.
+//
+// dir should be "" to list the root, and should not have
+// trailing slashes.
+//
+// This should return ErrDirNotFound if the directory isn't
+// found.
+func (f *Fs) List(ctx context.Context, dir string) (entries fs.DirEntries, err error) {
+	directoryID, err := f.dirCache.FindDir(ctx, dir, false)
+	if err != nil {
+		return nil, err
+	}
+	var iErr error
+	_, err = f.listAll(ctx, directoryID, false, false, func(info *api.Item) bool {
+		remote := path.Join(dir, info.Attributes.Name)
+		if info.Attributes.IsFolder {
+			// cache the directory ID for later lookups
+			f.dirCache.Put(remote, info.ID)
+			d := fs.NewDir(remote, time.Time(info.Attributes.ModifiedTime)).SetID(info.ID)
+			entries = append(entries, d)
+		} else {
+			o, err := f.newObjectWithInfo(ctx, remote, info)
+			if err != nil {
+				iErr = err
+				return true
+			}
+			entries = append(entries, o)
+		}
+		return false
+	})
+	if err != nil {
+		return nil, err
+	}
+	if iErr != nil {
+		return nil, iErr
+	}
+	return entries, nil
+}
+
+// FindLeaf finds a directory of name leaf in the folder with ID pathID
+func (f *Fs) FindLeaf(ctx context.Context, pathID, leaf string) (pathIDOut string, found bool, err error) {
+	// Find the leaf in pathID
+	found, err = f.listAll(ctx, pathID, true, false, func(item *api.Item) bool {
+		if item.Attributes.Name == leaf {
+			pathIDOut = item.ID
+			return true
+		}
+		return false
+	})
+	return pathIDOut, found, err
+}
+
+// CreateDir makes a directory with pathID as parent and name leaf
+func (f *Fs) CreateDir(ctx context.Context, pathID, leaf string) (newID string, err error) {
+	//fs.Debugf(f, "CreateDir(%q, %q)\n", pathID, leaf)
+	var resp *http.Response
+	var info *api.ItemInfo
+	opts := rest.Opts{
+		Method:       "POST",
+		Path:         "/files",
+		ExtraHeaders: map[string]string{"Accept": "application/vnd.api+json"},
```

Comment:
If you have this header for all calls then you can add it into the rest.Client just once with `f.srv.AddHeader(...)`

Reply:
It's not used for upload and download (and these won't work with that header) so I opted to only add it for the calls that really need it.

Reply:
OK that is fine.